server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2076,6 +2076,13 @@ public IndexMetadata build() {
                 );
             }
 
+            var aliasesMap = aliases.build();
+            aliasesMap.forEach((key, value) -> {
+                if (value.alias().equals(index)) {
+                    throw new IllegalArgumentException("alias name [" + index + "] and index name may not be the same");
```

Comment:
Can we make this a strong message about "conflict":

```suggestion
                    throw new IllegalArgumentException("alias name [" + index + "] self-conflicts with index name");
```